### DISCUSSION TITLE
ESP32 Support

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#3726] [Q&A]
Hi,

I see that slint has ESP32 support, and I'd like to understand which devices are covered and how. I don't think I'd be able to adapt slint to a board by programming that myself, therefore I'm only interested in the support that's currently built in.

Is any ESP32 based device with display would work with with the current support? Or only ESP32-S2 and ESP32-S3 like I think I understand from the docs?
If only ESP32-S2/S3 then would any board based on those ESP versions would work (is the display standard across all those boards)? For example, this is a board from AliExpress - https://www.aliexpress.com/item/1005005931825509.html, should it work?

Thanks

**Comment by tronical** [comment]
This looks like a powerful board. If you're programming with C++ and want to reuse a much existing platform abstraction as possible, then you could try this: https://components.espressif.com/components/slint/slint . It builds on Espressif's BSP.

What programming language are you targeting?

**Reply by yanshay** [reply]
I plan to use Rust for the applicative code at least. I don't mind if underneath I bring some C++ code that I don't need to deal with myself, or only minor changes.

I actually wanted a lower end board but couldn't find something as integrated and cheap that works with Slint so I'm thinking abount ESP32-S3.
For example, I looked initially at https://www.aliexpress.com/item/1005005938861565.html but it's not ESP32-S3, only ESP32 if I get it correctly.

Before I order I just want to know what board would work for me.

Are you saying that to use ESP32 with Slint programming has to be done with C++?

**Reply by tronical** [reply]
I think the ESP32-S3 is a very good starting point, from a software and hardware point of view.

> Are you saying that to use ESP32 with Slint programming has to be done with C++?

No, you can definitely use just Rust. That's what we did with the [S3 Box](https://github.com/slint-ui/slint/tree/master/examples/mcu-board-support#esp32-s3-box). But we don't have an out-of-the-box package or abstraction for all ESP32-S3 devices for Rust. You'll have to see what crate you can use to access drive the display and read from the touch screen in your case.

The esp-idf C++ approach takes care of some of what. In theory it can also be combined with Rust, but that is something we haven't tried yet.

I think at the end of the day you'll have to take something like our [S3 Box adaptation](https://github.com/slint-ui/slint/blob/master/examples/mcu-board-support/esp32_s3_box.rs), read it carefully, try to understand it, and then adapt it to your specific device and you specific PIN settings. You're not starting from scratch, but it's also not an out-of-the-box solution.

**Comment by yanshay** [comment]
Thanks, I'll try to see if I can handle this, but it doesn't look simple to do.

All those boards provide Lvgl and Arduino support (see http://pan.jczn1688.com/1 for what comes with this and similar boards), so if you could provide some integration on top of their abstractions or guidelines how to develop based on those configurations, you would get wide coverage for many boards with one time effort.

Because Lvgl Rust support is pretty poor from what I could see, and many want to move away from C++ to rust, I think that would drive many towards slint.

**Reply by tronical** [reply]
You're right. At the moment we merely have https://slint.dev/docs/rust/slint/docs/mcu/ as a general guide. But it does make sense to collect specific integrations over time, or improve our documentation. If you have concrete ideas, please let us know :)

**Comment by AllanOricil** [comment]
I would like to migrate from LVGL 8 to Slint on my pet project. I'm using platform.io to manage my esp32 project, which targets Esp32 CYD. Is there a guide to use slint with esp32-CYD and platform.io?

**Reply by ogoffart** [reply]
Depends if you want to use C++ or Rust.
There is no guide for the esp32-CYD specifically that I am aware of.
But the generic instructions should work https://docs.slint.dev/latest/docs/cpp/mcu/esp_idf or https://slint.dev/docs/rust/slint/docs/mcu/
